lib/services/elasticsearch.js

```diff
@@ -153,1195 +168,1048 @@ class ElasticSearch extends Service {
-        }
-
-        // set missing index & type if possible and add metadata
-        let lastAction; // NOSONAR
-
-        for (let i = 0; i < esRequest.body.length; i++) {
-          const item = esRequest.body[i];
-          const action = Object.keys(item)[0];
-
-          if (actionNames.indexOf(action) !== -1) {
-            lastAction = action;
-
-            if (!item[action]._type && esRequest.type) {
-              item[action]._type = esRequest.type;
-            }
-
-            if (!item[action]._type) {
-              return Bluebird.reject(this.getError(
-                'missing_data_collection_argument')
-              );
-            }
-
-            if (!item[action]._index && esRequest.index) {
-              item[action]._index = esRequest.index;
-            }
-
-            if (!item[action]._index) {
-              return Bluebird.reject(this.getError(
-                'missing_data_index_argument')
-              );
-            }
-
-            if (! (esCache[item[action]._index] &&
-              esCache[item[action]._index].includes(item[action]._type))) {
-              return Bluebird.reject(this.getError(
-                'index_or_collection_does_not_exist',
-                item[action]._index, item[action]._type)
-              );
-            }
-
-            if (item[action]._index === this.kuzzle.internalEngine.index) {
-              return Bluebird.reject(
-                this.getError('index_protected',
-                  this.kuzzle.internalEngine.index)
-              );
-            }
-          } else if (lastAction === 'index' || lastAction === 'create') {
-            item._kuzzle_info = kuzzleMetaCreated;
-          } else if (lastAction === 'update') {
-            // we can only update metadata on a partial update, or on an upsert
-            for (const prop of ['doc', 'upsert']) {
-              if (_.isPlainObject(item[prop])) {
-                item[prop]._kuzzle_info = kuzzleMetaUpdated;
-              }
-            }
-          }
-        }
+    return this.getMapping(index, collection, true)
+      .then(collectionMappings => {
 
-        return this.client.bulk(esRequest);
-      })
-      .then(response => this.refreshIndexIfNeeded(esRequest, response))
-      .then(result => {
-
-        // If some errors occured during the Bulk, we send a "Partial Error" response :
-        if (result.errors) {
-          result.partialErrors = [];
-          const items = [];
-          let row;
-
-          while ((row = result.items.shift()) !== undefined) {
-            const
-              action = Object.keys(row)[0],
-              item = row[action];
-
-            if (item.status >= 400) {
-              item.action = action;
-              result.partialErrors.push(item);
-            }
-            else {
-              items.push(row);
-            }
-          }
+        this._checkMappings(mappings);
 
-          result.items = items;
+        esRequest.body = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)